tutorials/create-an-example-application.md

```diff
@@ -0,0 +1,473 @@
+
+To run this example, first create the directory `/tmp/dqlite-data` with the following command:
+
+```bash
+mkdir -p /tmp/dqlite-data
+```
+
+Then, use the following command from within your project folder to run the application:
+
+```bash
+go run go-dqlite-demo.go
+```
+
+This should produce an output similar to the following:
+```bash
+ubuntu@dqlite-tutorial:~/go-dqlite-demo$ go run go-dqlite-demo.go 
+2023/09/11 14:40:46 App created
+2023/09/11 14:40:46 Database created
+2023/09/11 14:40:46 Table created
+```
+
+[note type="important" status="Info"]
+ Rerunning this app will result in an error since the table already exists. 
+[/note]
+
+Congratulations! You have created your first Dqlite application in Go.
+
+### Add Clustering
+
+Since Dqlite is a distributed database, it makes sense to have multiple nodes (a cluster) spun up. 
+A cluster of nodes can simply be created by providing the IP address of the leader node when creating a new node (thus, joining a cluster).
+In the previous example, we hard-coded the address directly in the code. Each node needs a unique IP address, so we need to make this value configurable.
+Let's use [cobra](https://github.com/spf13/cobra) to turn the previous example into an CLI application to which we can pass flags.
+
+[note type="important" status="Info"]
+ An introduction on cobra is out of scope for this tutorial. Please visit the [official documentation](https://cobra.dev/#getting-started) for more information on this library.
+[/note]
+
+Install the dependency with:
+```bash
+go get -u github.com/spf13/cobra/cobra
+```
+
+Now, move the code into a cobra command, expose the `--db` and `--join` flags to set the unique node address and other nodes in the cluster, respectively. 
+
+```Go
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"path/filepath"
+
+	"github.com/canonical/go-dqlite/app"
+	"github.com/spf13/cobra"
+	"golang.org/x/sys/unix"
+)
+
+func main() {
+	var db string
+	var join *[]string
+	var dir string
+
+	cmd := &cobra.Command{
+		Use:   "dqlite-demo",
+		Short: "Demo application using Dqlite",
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			dir := filepath.Join(dir, db)
+			if err := os.MkdirAll(dir, 0755); err != nil {
+				return fmt.Errorf("can't create %s: %v", dir, err)
+			}
+
+			// Set own address and specify all existing nodes in the cluster.
+			options := []app.Option{app.WithAddress(db), app.WithCluster(*join)}
+
+			// Setup Dqlite application
+			app, err := app.New(dir, options...)
+			if err != nil {
+				return err
+			}
+			log.Println("App created")
+
+			// Create a new database 'my-database' or just opens it if
+			// it already exists.
+			db, err := app.Open(context.Background(), "my-database")
+			if err != nil {
+				log.Fatal(err)
+			}
+			log.Println("Database created")
+
+            			http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I think the the http handler code is only introduced in the next step of the example?

Reply:
true - thanks for catching that!